### PR TITLE
Add client DSN setting

### PR DESCRIPTION
**Pull request by AlmightyYakob**
This adds the separate client DSN setting. So the plugin initializes the Python Sentry SDK with the backend (server) DSN, and the front-end initializes the JS Sentry SDK with the front-end (client) DSN.

### CODE REVIEW

File: plugins/sentry/girder_sentry/settings.py [hunk 1]
```diff
@@ -3,15 +3,22 @@
 
 
 class PluginSettings(object):
-    DSN = 'sentry.dsn'
+    _FRONTEND_DSN = 'sentry.frontend_dsn'
```
Comment (brianhelba) [inline plugins/sentry/girder_sentry/settings.py]: We want these to be public symbols, as all other plugin settings are.

Sorry that you have to fight with the public symbols test. 🙁

**Comment by AlmightyYakob** [comment]
@brianhelba This should be good now

File: plugins/sentry/girder_sentry/web_client/views/ConfigView.js [hunk 2]
```diff
@@ -8,27 +8,43 @@ import '../stylesheets/configView.styl';
...
+        },
+        'submit #g-sentry-client-form': function (event) {
+            event.preventDefault();
+            this.$('#g-sentry-client-error-message').empty();
+
+            this._saveSettings([{
+                key: 'sentry.frontend_dsn',
+                value: this.$('#g-sentry-client-dsn').val().trim()
+            }], {
+                error_target: '#g-sentry-client-error-message'
```
Comment (danlamanna) [inline plugins/sentry/girder_sentry/web_client/views/ConfigView.js]:
```suggestion
                errorTarget: '#g-sentry-client-error-message'
```

File: plugins/sentry/girder_sentry/web_client/views/ConfigView.js [hunk 3]
```diff
@@ -63,7 +79,7 @@ var ConfigView = View.extend({
                 timeout: 4000
             });
         }).fail((resp) => {
-            this.$('#g-sentry-error-message').text(
+            this.$(params.error_target).text(
```
Comment (danlamanna) [inline plugins/sentry/girder_sentry/web_client/views/ConfigView.js]:
```suggestion
            this.$(params.errorTarget).text(
```

File: plugins/sentry/girder_sentry/web_client/views/ConfigView.js [hunk 4]
```diff
@@ -8,27 +8,43 @@ import '../stylesheets/configView.styl';
...
+            this.$('#g-sentry-server-error-message').empty();
 
             this._saveSettings([{
-                key: 'sentry.dsn',
-                value: this.$('#g-sentry-dsn').val().trim()
-            }]);
+                key: 'sentry.backend_dsn',
+                value: this.$('#g-sentry-server-dsn').val().trim()
+            }], {
+                error_target: '#g-sentry-server-error-message'
```
Comment (danlamanna) [inline plugins/sentry/girder_sentry/web_client/views/ConfigView.js]:
```suggestion
                errorTarget: '#g-sentry-server-error-message'
```

File: plugins/sentry/girder_sentry/settings.py [hunk 5]
```diff
@@ -3,15 +3,22 @@
...
-@setting_utilities.validator(PluginSettings.DSN)
+@setting_utilities.validator(PluginSettings.FRONTEND_DSN)
+@setting_utilities.validator(PluginSettings.BACKEND_DSN)
 def _validateDsn(doc):
     if not doc['value']:
-        raise ValidationException('Sentry DSN must not be empty.', 'value')
+        raise ValidationException('DSN must not be empty.', 'value')
+
+    import re
+    if re.match(r'[\da-z]+@sentry.io/\d+', doc['value']) is None:
```
Comment (danlamanna) [inline plugins/sentry/girder_sentry/settings.py]: I don't think we need to do this extended validation.

**Comment by danlamanna** [comment]
Where in the javascript does this get initialized?

**Comment by AlmightyYakob** [comment]
> Where in the javascript does this get initialized?

https://github.com/girder/girder/blob/af53f40a465cd36188d21ca685bfaf6f92ae1843/plugins/sentry/girder_sentry/web_client/main.js#L13

**Comment by brianhelba** [comment]
@AlmightyYakob @danlamanna What's the status on this PR?

**Comment by brianhelba** [comment]
Looks like this is waiting for @AlmightyYakob to address feedback from @danlamanna .

**Comment by zachmullen** [comment]
This is in the release milestone, but IMO the release shouldn't be blocked on this. It's fine if we are going to merge this today, otherwise I'd suggest removing it from the milestone.

**Pull request merged**